apps/io-wallet-web-func/src/wallet-instance.ts

```diff
@@ -65,3 +80,54 @@ export const getCurrentWalletInstance: (
         ),
       ),
     );
+
+export const insertWalletInstance: (
+  walletInstance: WalletInstanceValid,
+) => RTE.ReaderTaskEither<WalletInstanceEnvironment, Error, void> =
+  (walletInstance) =>
+  ({ walletInstanceRepository }) =>
+    walletInstanceRepository.insert(walletInstance);
+
+export const revokeUserWalletInstancesExceptOne: (
+  userId: WalletInstance["userId"],
+  walletInstanceId: WalletInstance["id"],
+) => RTE.ReaderTaskEither<WalletInstanceEnvironment, Error, void> =
+  (userId, walletInstanceId) =>
+  ({ walletInstanceRepository }) =>
+    pipe(
+      walletInstanceRepository.getAllByUserId(userId),
+      TE.map(
+        O.fold(
+          () => [],
+          flow(
+            RA.filterMap((walletInstance) =>
+              walletInstance.id !== walletInstanceId
+                ? O.some(walletInstance.id)
+                : O.none,
+            ),
+          ),
+        ),
+      ),
+      TE.chain((walletInstancesId) =>
+        walletInstancesId.length
+          ? walletInstanceRepository.batchPatch(
+              walletInstancesId.map((id) => ({
+                id,
+                operations: [
+                  {
+                    op: "replace",
+                    path: "/isRevoked",
+                    value: true,
+                  },
+                  {
+                    op: "add",
+                    path: "/revokedAt",
+                    value: new Date(),
+                  },
+                ],
+              })),
+              userId,
+            )
+          : TE.right(void 0),
```

Comment:
What do you think about dividing this code into parts? It seems too long to me, you could split it into `getNonRevokedWalletInstanceIds`, `createPatchOperations` and `revokeWalletInstances`. It's just an idea.

Reply:
look at 0167d6cb0a700bfbd9936d2ef258fc91a097a365

Reply:
look at 4e3d943585b309c32328145175089deb16baf0af